dotNETStandard/HealthVaultProviderManagementPortal/HealthVaultProviderManagementPortal/Controllers/GoalsController.cs

```diff
@@ -30,26 +31,40 @@ public class GoalsController : Controller
         /// </summary>
         public async Task<ActionResult> Index(Guid personId, Guid recordId)
         {
-            var response = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.GetGoalsAsync(), personId, recordId);
-            return View(response);
+            var goalsResponse = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.Goals.GetActiveAsync(), personId, recordId);
+            var goalRecommendationsResponse = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.GoalRecommendations.GetAsync(), personId, recordId);
+
+            var model = new GoalsModel
+            {
+                Goals = goalsResponse.Goals,
+                GoalRecomendations = goalRecommendationsResponse.GoalRecommendations
+            };
+
+            return View(model);
         }
 
         /// <summary>
         /// Get a goal for the logged in user
         /// </summary>
         [HttpGet]
-        public async Task<ActionResult> Goal(Guid id, Guid personId, Guid recordId)
+        public async Task<ActionResult> Goal(Guid? id, Guid personId, Guid recordId)
         {
-            var response = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.GetGoalByIdAsync(id.ToString()), personId, recordId);
-            return View(response);
+            if (id.HasValue)
+            {
+                var response = await ExecuteMicrosoftHealthVaultRestApiAsync(api => api.Goals.GetByIdAsync(id.ToString()), personId, recordId);
+                return View(response);
+            }
+
+            var goal = new Goal();
+            return View(goal);
         }
 
         /// <summary>
         /// Update a goal for the logged in user
         /// </summary>
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public async Task<ActionResult> Goal(Goal goal, Guid personId, Guid recordId)
+        public async Task<ActionResult> Goal(Guid? id, Goal goal, Guid personId, Guid recordId)
```

Comment:
This is a form post in a website. the same page is used for create and update, depending on whether there's an ID